tools/src/test/java/org/apache/kafka/tools/GetOffsetShellTest.java

```diff
@@ -52,20 +55,30 @@ public class GetOffsetShellTest {
     private final int topicCount = 4;
     private final int offsetTopicPartitionCount = 4;
     private final ClusterInstance cluster;
+    private final String topicName = "topic";
 
     public GetOffsetShellTest(ClusterInstance cluster) {
         this.cluster = cluster;
     }
 
     private String getTopicName(int i) {
-        return "topic" + i;
+        return topicName + i;
     }
 
-    public void setUp() {
+    @BeforeEach
+    public void before(TestInfo testInfo) {
```

Comment:
Hmm, this is a weird test to verify this behavior. @hachikuji @jolshan had identified a test that seemed to be a better candidate.

Reply:
I'm not sure what _weird test_ you mean here. I've refactored it by using `serverProperties` to provide custom server properties for specific tests, if that's what you meant _weird_. Thanks.

Reply:
Hey sorry for the delay. I made the following modification to ListOffsetsIntegrationTest. I also added the compressionType argument to produceMessages in TestUtils.

```
  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
  @ValueSource(strings = Array("zk", "kraft"))
  def testMaxTimestampOffset(quorum: String): Unit = {
    val maxTimestampOffset = runFetchOffsets(adminClient, OffsetSpec.maxTimestamp())
    assertEquals(1, maxTimestampOffset.offset())
  }

...

  def produceMessages(): Unit = {
    val records = Seq(
      new ProducerRecord[Array[Byte], Array[Byte]](topicName, 0, 100L,
        null, new Array[Byte](10)),
      new ProducerRecord[Array[Byte], Array[Byte]](topicName, 0, 999L,
        null, new Array[Byte](10)),
      new ProducerRecord[Array[Byte], Array[Byte]](topicName, 0, 200L,
        null, new Array[Byte](10)),
    )
    TestUtils.produceMessages(brokers, records, -1, compressionType = "gzip")
  }
```

Unfortunately produceMessages is called in before each, so it isn't as easily parameterized. But this had a few lines changes and reproduced the error.

Reply:
We should also fix this test to use 10 bytes rather than 10KB since it was hiding an error as well.

Reply:
@showuon As Justine pointed out, we have a test that verifies that list offsets requests works correctly. That's a better test to verify this than the offset shell test.

Reply:
> We should also fix this test to use 10 bytes rather than 10KB since it was hiding an error as well.

Reply:
`producer.flush` is fine if we set large `batch.size` and `linger.ms` so we can control when it happens.

Reply:
I don't think there is a need to have 10KB records regardless -- we can't have this size in a single batch consistently.

So while it is ok to configure via the flush method -- let's make the record size smaller. We should confirm the batching is as expected when we run the tests. :)

Reply:
> So while it is ok to configure via the flush method -- let's make the record size smaller. We should confirm the batching is as expected when we run the tests. :)

+1 to this comment 👍

Reply:
Thanks all for the comments! I've updated the tests. Please take a look when available. Thanks.